### ISSUE TITLE
[bug] Access Policy: Contexts

### ISSUE THREAD

**Issue by Ruslan-Aleev** [modxcms/revolution#14407]
### Summary
Recently we tested all the settings in the **Access Policy** and grouped them into logical groups, but almost every setting has an inaccuracy or even a bug.

**Guys, we need help in testing these problems, both from the MODX-user and from the MODX-components.**

_"Contexts" Group:_
- **view_context** - Enable / Disable section with "Gear-> Contexts", for a direct link does not work. If in the tree to go to context editing via the context menu, the page hangs, but does not give an error. Probably a bug.
- **edit_context** - Prevent editing context, there are no comments.
- **save_context** - Doesn't work, contexts can both be edited, saved and create new ones. Probably a bug.
- **delete_context** - To delete or remove any Context. Description Question: Why delete and remove, is it not the same thing?
- **new_context** - Prevent creating context. You can click create, fill all, but only at the save stage, a message with access error will pop up. It is more logical that the access error pops up when you click on the create button.

### Environment
MODX >2.x
https://github.com/modxcms/revolution/issues/14405, https://github.com/modxcms/revolution/issues/14406

**Comment by Ibochkarev** [comment]
Re-checked this on current 3.x (`settings_version` 3.3.0-dev, this tree plus live `modx_menus` / Administrator and Developer policies). All five keys are still on Administrator and Developer. Content Editor has none of them. The 2019 bugs are still there. Grid buttons got better after #16653. `save_context` is still unused in PHP.

## What each key does today

| Permission | Where it is enforced | What it is *not* |
|---|---|---|
| `view_context` | Gear → Contexts menu (`modx_menus` action `context`). List controller. View controller. `Context/Get`, `GetList` | Does not show the Gear parent by itself (parent `admin` requires `settings`). Does not gate tree “Edit Context”. Does not gate `Context/Update` |
| `edit_context` | Update controller. `Context/Update`, `UpdateFromGrid`. Tree `pedit` (“Edit Context”). Combined with `save_context` for grid edit. TV media-source grid also wants this pair | Does not hide Gear → Contexts. Does not hide the Save button on the update page |
| `save_context` | ExtJS only: grid `userCanEdit` / `userCanCreate`, TV sources `editContexts` | **No processor checks this.** Create, Duplicate, Update, UpdateFromGrid ignore it |
| `delete_context` | `Context/Remove`. Tree `pdelete` (“Remove Context”). Grid delete. Core contexts (`mgr`, `web`, …) cannot be deleted | Lexicon still says “delete or remove” |
| `new_context` | `Create`, `Duplicate`. Tree `pnew` is Duplicate Context, not Create. Grid New/Duplicate hidden unless `new_context` **and** `save_context`. Update-page Duplicate looks at `MODx.perm.new_context` only | Create processor still only checks `new_context` |

Context Settings CRUD (`Context/Setting/*`) uses `settings`, not these five keys. The update page also loads an ACL grid. Object-level `modContext` policies (`view` / `save` / `remove`) sit on top of the manager keys.

## Versus the 2019 notes

**`view_context`** — The list page and `?a=context` now require `view_context`. The leftover problem is the resource tree. `Resource/GetNodes` puts `pedit` on a context node from `edit_context` alone. Tree → Edit Context loads `?a=context/update` (controller: `edit_context`). The panel then calls `Context/Get` (`view_context`) with `useLoadingMask: true` and **no failure listener**:

```js
MODx.Ajax.request({
    params: { action: 'Context/Get', key: this.config.context },
    listeners: { success: { fn: function (response) { /* … */ } } }
});
```

Without `view_context` the request fails, the mask stays, and you get no error. That is the hang in the original report.

Gear parent is `settings`. `header.php` skips the whole parent, so `view_context` without `settings` never shows Contexts. Same parent-vs-child split as Access → Users in #14406. Related: #16595.

**`edit_context`** — This is the real save gate today. Turn it off and Update / UpdateFromGrid deny. The grid still wants `edit`+`save` for inline edit. The update page Save button does not look at `save_context`.

**`save_context`** — Confirmed dead on the server. Turn it off, keep `edit_context` / `new_context`: grid New/Edit hide after #16653, tree Edit/Duplicate still work, update page still saves. That is why “contexts can be edited, saved and created.”

**`delete_context`** — Works. “To delete or remove any Contexts” is the same tautology as `delete_user` / `delete_propertyset`. Tree label is already “Remove Context”; the processor is `Remove`. No second action.

**`new_context`** — Partly improved. The grid hides Create/Duplicate without `new`+`save` (#16653). The create window posts `Context/Create`, which only checks `new_context`. Tree Duplicate and the update-page Duplicate button only check `new_context`. There is no create-context item in the resource tree.

## Suggested shape

Keep view / new / edit / save / delete. Put `save_context` on the processors that actually write.

1. **Menu + list page = `view_context`.** Gear parent vs `settings` is #16595. Do not make Contexts depend on `settings` to appear.

2. **Tree Edit Context needs `view_context` and `edit_context`.** Hide `pedit` unless both are true, so you never open the hanging page. Require both on the update controller as well.

3. **`Context/Update` and `UpdateFromGrid` should require `save_context`** (keep `edit_context` on the page). That matches users: page=`edit_*`, processor=`save_*`. Hide the update-page Save button without `save_context`.

4. **`Create` / `Duplicate` should require `new_context` and `save_context`**, matching the grid. The update-page Duplicate button should use the same pair.

5. **`setup()` in `modx.panel.context.js` needs a `failure` listener** that hides the mask and shows the processor message. Do this even if tree/`Get` permissions are aligned.

6. **`delete_context` = permanently delete contexts.** Leave core contexts undeletable.

7. **Lexicons (en + ru at minimum)**
   - `delete_context`: “Permanently delete contexts.”
   - `save_context`: “Save context name, description, rank, and (with `settings`) context settings.”
   - `view_context` / `edit_context`: list page vs update page, and that tree Edit uses both.

Happy to turn this into a PR if maintainers want that split. Same family as #14405 and #14406.
